JitsiConference.js

```diff
@@ -304,6 +305,16 @@ JitsiConference.prototype.join = function(password) {
 };
 
 /**
+ * Creates a new {@link JitsiAuthConnection} which authenticates and upgrades
+ * the user's role to moderator.
+ *
+ * @returns {JitsiAuthConnection}
+ */
+JitsiConference.prototype.createAuthenticationConnection = function() {
```

Comment:
1. I do not support multiple names for one and the same thing because that makes it difficult for me comprehend the source code. Is the abstraction called `AuthenticationConnection` or `JitsiAuthConnection`?

2. I find the abstraction too complex:

* As an API client/consumer, I have to `jitsiConference.createAuthenticationConnection()`, then call `authenticateAndUpgradeRole` on the return value, and then wait on the returned `Promise`.

* I understand I need an object to call `cancel` on.

* `cancel` renders the `JitsiAuthConnection` instance un-re-usable because `cancel` only sets `canceled` and `authenticateAndUpgradeRole` only returns `_promise`.

* I'd suggest having `JitsiConference.prototype.authenticateAndUpgradeRole` which returns a Promise with a `cancel` method (which should be documented, of course). This way there will be no need for creating an object and then using it to only get another object. Then `cancel` may be internally implemented nicely with `reject` in order to satisfy the documentation that `cancel` always rejects with an empty object.

* Alternatively, I'd drop the whole `JitsiAuthConnection ` and `Promise` business because of the weirdness with returning a `Promise` and having a `onLoginSuccessful` callback which @saghul pointed out. If there's going to be callback involved anyway, then I'd have `JitsiConference.prototype.authenticateAndUpgradeRole({ onLoginSuccessful, onError/Rejected, onFullfilled/Success})` i.e. a function that's deals with callbacks only. Then if the caller wants cancellation as well, they may as well add a `cancelled` callback in there, internally we can invoke it whenever we currently do `this.canceled`, and then we don't even have to promise that we'll do `reject` with an empty object because it's the API client/consumer who's doing the cancellation anyway.